commands.go

```diff
@@ -521,20 +521,25 @@ func banCMD(line string, u *User) {
 		return
 	}
 	// check if the ban is for a certain duration
+	var dur time.Duration = 0
 	if len(split) > 1 {
-		dur, err := time.ParseDuration(split[1])
+		var err error
+		dur, err = time.ParseDuration(split[1])
 		if err != nil {
 			u.room.broadcast(Devbot, "I couldn't parse that as a duration")
 			return
 		}
-		victim.ban(u.Name + " has been banned by " + u.Name + " for " + dur.String())
+	}
+	if dur != 0 {
+		victim.ban(victim.Name + " has been banned by " + u.Name + " for " + dur.String())
 		go func(id string) {
 			time.Sleep(dur)
 			unbanIDorIP(id)
 		}(victim.id) // evaluate id now, call unban with that value later
-		return
 	}
-	victim.ban(u.Name + " has been banned by " + u.Name)
+	id := victim.id
+	victim.ban(victim.Name + " has been banned by " + u.Name)
```

Comment:
Maybe the `user.ban` func should always use `kickID`

Reply:
What do you mean?

Reply:
since the ban function kicks the banned user, it should maybe use the new kick command to kick all users with the same id as the banned user

Reply:
As of [7bd2c1a](https://github.com/quackduck/devzat/pull/146/commits/7bd2c1aeeca1d735403a3fd2e09de8ad75d966ef), it is what is done at line 542 of commands.go.